util/src/main/java/com/redhat/lightblue/util/JsonUtils.java

```diff
@@ -443,6 +443,33 @@ private static void indent(StringBuilder bld, int depth) {
         }
     }
 
+    /**
+     * An approximation of how much space will a JsonNode take in memory. Does not take jvm optimizations
+     * or architecture into account. Useful only to compare JsonNodes with each other.
+     *
+     * @param node
+     * @return
+     */
+    public static int size(JsonNode node) {
+        int size = 0;
+        if (node instanceof ArrayNode) {
+            for (Iterator<JsonNode> elements = ((ArrayNode) node).elements(); elements.hasNext();) {
+                size += size(elements.next());
+            }
+        } else if (node instanceof ObjectNode) {
+            for (Iterator<Map.Entry<String, JsonNode>> fields = ((ObjectNode) node).fields(); fields.hasNext();) {
+                Map.Entry<String, JsonNode> field = fields.next();
+                size += field.getKey().length();
+                size += size(field.getValue());
+            }
+        } else if (node instanceof NumericNode) {
+            size += 4;
+        } else {
+            size += node.asText().length();
```

Comment:
No, but null can be passed to JsonUtils.size() in this context, and this is the only place it'll throw a NPE. Null was not a concern where this was copied from.